File: frontend/src/app/services/cliente.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Cliente {
  id: number;
  ruc: string;
  razonSocial: string;
  telefono: string;
  correo: string;
  direccion: string;
}

@Injectable({
  providedIn: 'root',
})
export class ClienteService {
  private apiUrl = 'https://localhost:7106/api/Clientes';

  constructor(private http: HttpClient) {}

  buscarClientes(term: string): Observable<Cliente[]> {
    let params = new HttpParams();
    if (term && term.trim() !== '') {
      params = params.set('search', term);
    }

    return this.http.get<Cliente[]>(this.apiUrl, { params });
  }
}

File: backend/ClientesApi/Models/Cliente.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

[Table("CLIENTES")]
public class Cliente
{
    [Column("ID")]
    public int Id { get; set; }

    [Column("RUC")]
    [MaxLength(20)]
    public string Ruc { get; set; }

    [Column("RAZON_SOCIAL")]
    [MaxLength(250)]
    public string RazonSocial { get; set; }

    [Column("TELEFONO")]
    [MaxLength(30)]
    public string Telefono { get; set; }

    [Column("CORREO")]
    [MaxLength(250)]
    public string Correo { get; set; }

    [Column("DIRECCION")]
    public string Direccion { get; set; }
}